client/cli/src/params/node_psk_key_params.rs

```diff
@@ -0,0 +1,119 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use clap::Args;
+use sc_network::config::NodePreShareKeyConfig;
+use std::{path::PathBuf, str::FromStr};
+
+use crate::{error};
+
+/// Parameters used to create the `NodePreSharedKeyConfig`, which determines the pre shared key
+/// used for libp2p networking.
+#[derive(Debug, Clone, Args)]
+pub struct NodePreSharedKeyParams {
+	/// Endpoint for requesting a pre shared key.
+	#[clap(long, value_name = "URL")]
+	pub rpc_addr_for_pre_shared_key: Option<String>,
```

Comment:
I think it's better to put this directly in `NetworkParams` and name it `listen_rpc`. Also, I think Option here is unnecessary, at least one address must be provided and we cannot set the default one.